code/modules/events/alien_infestation.dm

```diff
@@ -50,7 +50,7 @@
 		message_admins("An event attempted to spawn an alien but no suitable vents were found. Shutting down.")
 		return MAP_ERROR
 
-	var/list/candidates = get_candidates("alien", null, ROLE_ALIEN)
+	var/list/candidates = pollCandidates("Do you want to play as an alien larva?", ROLE_ALIEN, null, ROLE_ALIEN, 100, POLL_IGNORE_ALIEN_LARVA)
```

Comment:
Well then the proc is not working properly, since it was grabbing ghosts instead of asking.